main.tf

```diff
@@ -178,7 +179,7 @@ data "template_file" "init_master" {
 
   vars {
     kubeadm_token = "${module.kubeadm-token.token}"
-    dns_name      = "${var.cluster_name}.${var.hosted_zone}"
+    dns_name      = "internal.${var.cluster_name}.${var.hosted_zone}"
```

Comment:
I think we should pass here both the internal as well as the external DNS names. We should use them both to generate the certificate SANs (in `init-aws-kubernetes-master.sh`, lines 89-91) and create for both of them kubeconfig  (same file, lines 107-114) so that you can pick the one with external access or internal access. But I can do this while merging this if you want.